src/oci_features.c

```diff
@@ -142,8 +142,10 @@ crun_features_add_seccomp_info (yajl_gen json_gen, const struct linux_info_s *li
   yajl_gen_map_open (json_gen);
 
   add_bool_to_json (json_gen, "enabled", linux->seccomp.enabled);
+#ifdef HAVE_SECCOMP
   add_array_to_json (json_gen, "actions", linux->seccomp.actions);
```

Comment:
would it be more robust if we test it at runtime `if (linux->seccomp.actions)` instead of using the preprocessor?  So it would catch also cases where seccomp is not configured, for any reason.

Reply:
Actually, that was my original patch, but then I wanted to make it consistent and used `ifdef`s in both places.

Reply:
oh, I misread your comment. I read `if (linux->seccomp.enabled)`
But yeah, checking for `seccomp.actions` is probably better. @giuseppe do you want to have separate check for `seccomp.operators` or one common check with `seccomp.actions` for both of them?

Reply:
let's make it separate

Reply:
ok, I looked at this again and am no longer sure the check you suggested is required. That's because when we `HAVE_SECCOMP`, crun will preallocate those arrays, and they will always resolve to a truthy value. Or am I missing something?

https://github.com/containers/crun/blob/4b75c7cb574d642481c9346eb3d54e91c33ebaf2/src/libcrun/container.c#L4068-L4073

Reply:
yes that is true, and your code would fix that.  I was suggesting to have a runtime check so that if we change these lines in future, the check will still be valid.

Reply:
e.g. imagine one day we change this code to not populate archs if `num_archs == 0`.  The runtime check will still be valid, while if we use `HAVE_SECCOMP` we will still try to access the array